doc/ch-fromhost_desc.rst

```diff
@@ -17,23 +17,45 @@ Description
 Inject files from the host into the Charliecloud image directory
 :code:`IMGDIR`.
 
-The purpose of this command is to inject files into a container image that are
-necessary to run the container on a specific host; e.g., GPU libraries that
-are tied to a specific kernel version. **It is not a general copy-to-image
-tool**; see further discussion on use cases below. It should be run after
-:code:`ch-convert` and before :code:`ch-run`. After invocation, the image is
-no longer portable to other hosts.
+The purpose of this command is to inject arbitrary files and loadable Libfabric
+shared object providers into a container that are necessary to access host
+specific resources; usually GPU or proprietary interconnets. **It is not a
+general copy-to-image tool**; see further discussion on use cases below.
+
+It should be run after:code:`ch-convert` and before :code:`ch-run`. After
+invocation, the image is no longer portable to other hosts.
 
 Injection is not atomic; if an error occurs partway through injection, the
-image is left in an undefined state. Injection is currently implemented using
-a simple file copy, but that may change in the future.
+image is left in an undefined state and should be re-unpacked from storage.
+Injection is currently implemented using a simple file copy, but that may
+change in the future.
+
+Arbitrary file and Libfabric injection are handled differently.
+
+Loadable Libfabric providers
+----------------------------
+
+Loadable Libfabric providers, e.g., files that end in :code:`-fi.so`, are copied
+to a sepcific directory, :code:`libfabric`, inside the image where
+:code:`libfabric.so` is found. If the directory doesn't exist it is created.
```

Comment:
Where is it created?

Reply:
The `libfabric` directory is created in the container at the path where the container `libfabric.so` is found.

Note that placing the loadable shared objects at the same path as the container `libsofabric.so` without the `libfabric` directory didn't work; it had to be in the directory. This was requirement was inferred by building loadable providers and noting their location relative to built-in providers. I didn't find anything in their documentation about this.